draft-duke-quic-load-balancers.md

```diff
@@ -59,270 +61,545 @@ author:
+   but the others MUST be global.
+
+   For Encrypted CID Routing, this consists of the Server ID,
+   Server ID Length, Key, and Connection ID Length. The Server ID
+   is unique to each server, but the others MUST be global.
+
+## Out of band sharing
+
+   When there are concerns about the integrity of the path between
+   load balancer and server, operators may share routing information
+   using an out-of-band technique, which is out of the scope of
+   this specification.
+
+   To simplify configuration, the global parameters can be shared
+   out-of-band, while the load balancer sends the unique server
+   IDs via the truncated message formats presented below.
+
+## QUIC-LB Message Exchange
+
+   QUIC-LB load balancers send the encoding parameters to servers
+   as they discover the servers, using a single packet to each that
+   resembles QUIC. They periodically retransmit this packet to each
+   server until that server responds with a QUIC-LB ack. Specifics
+   of this retransmission are implementation-dependent.
+
+   These message formats are specific to QUICv2 and experimental
+   versions leading up to QUICv2. They may require revision for
+   future versions of QUIC.
+
+## Packet Header Format
+   QUIC-LB messages resemble QUIC packets with a long header. They are
+   sent when a load balancer boots up, or detects a new server in the
+   pool. QUIC-LB packets are delivered in a UDP datagram.
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+
+| Type = 0xfb   |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                      	Version (32)                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|      0x00     |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                           Length (i)                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|     Random    |
++-+-+-+-+-+-+-+-+
+{: #quic-lb-header title="QUIC-LB Header"}
+
+   The type field is 0xfb, which is otherwise unused in QUICv2.
+
+   The Version field allows QUIC-LB to use the Version Negotiation
+   mechanism. All messages in this specification are specific to
+   QUICv2, as future QUIC versions may use the 0xfb type for other
+   purposes. Therefore, the Version field should be set as 0xff00000d,
+   the current QUIC version.
+
+   Load balancers MUST cease sending QUIC-LB packets of this version
+   to a server when that server sends a Version Negotiation packet
+   that does not advertise the version.
+
+   The 0x00 byte indicates that there are no connection IDs present
+   in the header.
+
+   The length field is a variable-length integer that encodes the
+   length of the entire QUIC-LB packet, in octets, after the length
+   field. The variable integer encoding is the same as that in
+   {{QUIC-TRANSPORT}}.
```

Comment:
Why do we need an explicit length, instead of just implicitly calculating it from the UDP payload size?